frontend/src/components/modals/NotifyUsersModal.tsx

```diff
@@ -0,0 +1,225 @@
+            .filter((rating) => rating.forCustomer === customer.id)
+            .map((e) => e.createdByUser);
+
+          if (representatives) {
+            representatives.forEach((rep) => {
+              if (!ratingsForTask.includes(rep.id))
+                if (!missingRepresentatives.some((e) => e.id === rep.id))
+                  missingRepresentatives.push({ ...rep, checked: false });
+            });
+          }
+        });
+      }
+
+      // Filter developers, check who haven't given ratings, add checked attribute for all of them
+      const ratingIds = task.ratings.map((rating) => rating.createdByUser);
+      const missingDevelopers = allUsers
+        ?.filter((user) => user.type === RoleType.Developer)
+        .filter((developer) => !ratingIds.includes(developer.id))
+        .map((dev) => ({ ...dev, checked: false }));
+
+      setMissingUsers([
+        ...(missingRepresentatives ?? []),
+        ...(missingDevelopers ?? []),
+      ]);
+    }
+  }, [allUsers, customers, task.ratings, task.roadmapId, userInfo]);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setIsLoading(true);
+
+    const checkedUsers = missingUsers?.filter((user) => user.checked);
+    // Parse RoadmapUsers from list of CheckedUsers
+    const usersToNotify = allUsers?.filter((user) =>
+      checkedUsers?.some((checkedUser) => checkedUser.id === user.id),
+    );
+
+    if (!usersToNotify || usersToNotify?.length === 0) {
+      setIsLoading(false);
+      return setErrorMessage(t('Please select at least one user'));
+    }
+
+    dispatch(roadmapsActions.notifyUsers({ users: usersToNotify, task })).then(
+      (res) => {
+        setIsLoading(false);
+        if (roadmapsActions.notifyUsers.rejected.match(res)) {
+          if (res.payload) {
+            setErrorMessage(res.payload.message);
+          }
+        } else {
+          closeModal();
+        }
+      },
+    );
+  };
+
+  const checkAll = (checked: boolean) => {
+    const copy = [...(missingUsers ?? [])];
+    setAllChecked(!allChecked);
+    if (!copy) return;
+    copy.forEach((_user, idx) => {
+      copy[idx].checked = checked;
+    });
+    setMissingUsers(copy);
+  };
+
+  const checkUser = (checked: boolean, idx: number) => {
+    setAllChecked(false);
+    const copy = [...(missingUsers ?? [])];
+    if (!copy) return;
+    copy[idx].checked = checked;
+    setMissingUsers(copy);
+  };
+
+  return (
+    <>
+      <Form onSubmit={handleSubmit}>
+        <ModalHeader>
+          <h3>
+            <Trans i18nKey="Send notification by email" />
+          </h3>
+        </ModalHeader>
+        <ModalContent>
+          <p className={classes(css.subtitle)}>
+            <Trans i18nKey="Send the notification to" />
+          </p>
+          <Checkbox
+            label="All"
+            checked={allChecked}
+            onChange={(checked: boolean) => checkAll(checked)}
```

Comment:
The types on the checkbox listeners can be left out due to type inference
```suggestion
            onChange={(checked) => checkAll(checked)}
```